### PR TITLE
Giving someone an object makes them pick it up as well.

### PR DESCRIPTION
**Pull request by BiancaWilkson**
## What Does This PR Do

This PR makes it so giving someone an item calls that item's pickup proc.
## Why It's Good For The Game

This causes a few bugs, including poisoned papers not transferring poisons when given to someone. This prevents similar oversights from happening in the future.

## Testing

Loaded in, tried giving a skrell some paper, realized I couldn't give something to a mob with no client, commented out that code, loaded in again, it worked.

I gave skrells a variety of items, including normal IDs, death nettles, and poisoned papers. All of it functioned as normal

:cl:
Tweak: Giving someone an item forces them to also pick it up.
Fix: Giving someone a poisoned piece of paper will transfer the poison to them.
/:cl:

### CODE REVIEW

File: code/modules/paperwork/paper.dm [hunk 1]
```diff
@@ -843,6 +843,15 @@
 			contact_poison = null
 	. = ..()
 
+/obj/item/paper/on_give(mob/living/carbon/giver, mob/living/carbon/receiver)
+	if(contact_poison && ishuman(receiver))
```
Comment (DGamerL) [inline code/modules/paperwork/paper.dm:847]: Could you make this an early return?

**Comment by Gatchapod** [comment]
I want to bring up a possibly much larger change.
Wouldn't a better way to handle this be making `on_give()` call `pickup()`? It's possible that this oversight was originally made because of the assumption this functionality already exists.
When I brought it up on Discord, Qwerty seemed to agree.

Is there any item we would particularly want to not work like that?

**Comment by BiancaWilkson** [comment]
> I want to bring up a possibly much larger change. Wouldn't a better way to handle this be making `on_give()` call `pickup()`? It's possible that this oversight was originally made because of the assumption this functionality already exists. When I brought it up on Discord, Qwerty seemed to agree.
> 
> Is there any item we would particularly want to not work like that?

This actually wouldn't be a large change at all, I'll push the changes and update the PR.

File: code/modules/paperwork/paper.dm [hunk 2]
```diff
@@ -840,14 +840,15 @@
...
-	if(contact_poison && ishuman(user))
-		var/mob/living/carbon/human/H = user
-		var/obj/item/clothing/gloves/G = H.gloves
-		if(!istype(G) || !G.safe_from_poison)
-			H.reagents.add_reagent(contact_poison, contact_poison_volume)
-			add_attack_logs(src, user, "Picked up [src], coated with [contact_poison] by [contact_poison_poisoner]")
-			contact_poison = null
-	. = ..()
+	..()
+	if(!(contact_poison && ishuman(user)))
```
Comment (DGamerL) [inline code/modules/paperwork/paper.dm:844]:
```suggestion
	if(!contact_poison || !ishuman(user))
```
I prefer inverting all statements like this, makes it look a bit clearer

**Comment by BiancaWilkson** [comment]
I did a fair bit of testing different possible cases and it all worked as I hoped it would. There didn't seem to be any unintended side-effects.

**Review by Burzah — APPROVED** [review]

**Review by DGamerL — APPROVED** [review]

**Review by Contrabang — APPROVED** [review]

**Pull request merged**